src/main.rs

```diff
@@ -197,12 +197,29 @@ fn actual_main(paths: ForemanPaths) -> ForemanResult<()> {
 
             let mut cache = ToolCache::load(&paths)?;
 
+            let mut accumulated_errors: HashMap<&String, ForemanError> = HashMap::new();
             for (tool_alias, tool_spec) in &config.tools {
                 let providers = ToolProvider::new(&paths);
-                cache.download_if_necessary(tool_spec, &providers)?;
+
+                match cache.download_if_necessary(tool_spec, &providers) {
+                    Ok(_) => {}
+                    Err(err) => {
+                        accumulated_errors.insert(tool_alias, err);
+                    }
+                }
                 add_self_alias(tool_alias, &paths.bin_dir())?;
```

Comment:
I think we want to move this call into the `Ok(_)` branch, since we don't want to create the self alias if we were not able to download it.

You can pattern-match it, or chain it with [`and_then`](https://doc.rust-lang.org/std/result/enum.Result.html#method.and_then).